nostr-client-plus/src/bin/aggregator.rs

```
@@ -403,6 +411,238 @@ async fn run() -> Result<()> {
     Ok(())
 }
 
+async fn finalize_classification(
+    assign_event_id: String,
+    payload: &ResultPayload,
+    event_id: String,
+    collection: &Collection<AssignerTask>,
+    finish_job_collection: &Collection<FinishedJobs>,
+) -> Result<()> {
+    tracing::debug!("Finalizing classifications for event: {}", event_id);
+    let filter = doc! {
+        "event_id": event_id.clone(),
+    };
+
+    let mut cursor = collection
+        .find(filter, None)
+        .await?;
+    let mut tasks: Vec<AssignerTask> = Vec::new();
+    while let Some(task) = cursor.try_next().await? {
+        tasks.push(task);
+    }
+
+    let mut processing_count = 0;
+    let mut answers: HashMap<String, i32> = HashMap::new();
+    for task in tasks.iter() {
+        tracing::debug!("Worker id: {}", hex::encode(task.worker.to_bytes()));
+        if task.result.is_empty() && task.status == AssignerTaskStatus::Pending {
+            processing_count += 1;
+            break;
+        }
+        let result_identifier = match task.get_result_identifier() {
+            Ok(val) => {
+                val
+            }
+            Err(err) => {
+                return Err(anyhow!("Failed to find result identifier for task: {}", task.event_id));
```

You are returning an error only if `Result` is `Err`, and the message reads "Failed to find a result identifier".
However, if `get_result_identifier` returns `None`, you don't have a valid identifier. Not only this is not reported, but a few lines below (`if let Some(identifier)..`), you have no `else` branch, so we basically silently fail.

The same is true when we check if the string is empty a few lines below.

So, first of all the match statement can take care of everything: branches should read like the following

```rust
match task.get_result_identifier() {
    Ok(Some(val)) => {
        if !identifier.is_empty() {
            // happy path code, the same you have below
        } else {
            // what we do if identifier is empty
        }
    Ok(None) => { // what we do when it's None }
    Err(err) => { // Do something when it's Err and, possibly print the error }
}
```